Tribler/Core/Session.py

```diff
@@ -406,9 +405,15 @@ def start(self):
         if self.upgrader_enabled:
             self.upgrader = TriblerUpgrader(self)
             self.readable_status = STATE_UPGRADING_READABLE
-            self.upgrader.run()
+            upgrader_deferred = self.upgrader.run()
+        else:
+            upgrader_deferred = succeed(None)
 
-        startup_deferred = self.lm.register(self, self.session_lock)
+        def after_upgrade(_):
+            self.upgrader = None
+
+        startup_deferred = upgrader_deferred.addCallbacks(lambda _: self.lm.register(self, self.session_lock),
+                                                          lambda _: None).addCallbacks(after_upgrade, lambda _: None)
```

Comment:
We should at least log calls to an `errBack`. I think there is a default `addCallback` we have implemented recently in Tribler. Using that one provides some logging 👍

Reply:
I'll see if the default `addCallback` works.